sw/device/boot_rom2/uart_log.c

```diff
@@ -0,0 +1,62 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0`
+
+#include "sw/device/boot_rom2/uart_log.h"
+
+#include "sw/device/lib/arch/device.h"
+#include "sw/device/lib/base/print.h"
+#include "sw/device/lib/runtime/hart.h"
+
+/**
+ * Handle for the 0th UART port.
+ */
+static dif_uart_t uart0;
+
+dif_uart_t *uart_log_handle(void) { return &uart0; }
+
+/**
+ * Tries to write a buffer to the 0th UART. This function only exists to
+ * drive logging, and should otherwise not be called directly.
+ *
+ * This function is called by error-handling code, and as such a failure
+ * within it is treated asa double-fault.
+ *
+ * @param ignored a data pointer required by buf_sink_t, which is ignored.
+ * @param buf a buffer to write to UART.
+ * @param len the length of the buffer.
+ * @return the number of bytes written, that is, the value of len.
+ */
+static size_t uart0_send_buf(void *ignored, const char *buf, size_t len) {
+  size_t total_len = len;
+  while (len > 0) {
+    size_t bytes_written;
+    bool success =
```

Comment:
Just `if (!dif_uart_bytes_send` here?

Reply:
I mostly wanted to write down the actual error type, so that there is some hope this might break once Silvestrs or Sam changes the error type to something more concrete.

But, also, I just don't like big function calls in if conditions if i can avoid them.